rfcs/0007-handling-corruption.md

```diff
@@ -0,0 +1,533 @@
+  `get` caller.
+
+The corruption error returned should be of a distinct type, allowing it to be easily
+identified as either non-transient corruption or a transient error (such as
+network connectivity issues).
+
+Since `get` calls may be ongoing during corruption, it is possible that the
+callback will be called continuously and by multiple threads. Therefore, the
+`DBOptions.on_corruption` callback must be thread-safe. Additionally, the callback should
+avoid slow or blocking operations to ensure the database remains available if
+corruption is encountered.
+
+The caller can choose to abort the application if a corruption error is
+returned by `get`. It is the responsibility of the caller to report corruption details
+to operators with detailed information on the corruption.
+
+#### Handling Manifest Corruption during Open
+If corruption is detected in a manifest when calling `DB::open()`, the
+callback `DBOptions.on_corruption` is invoked. Regardless of the return value
+of the callback, `DB::open()` will return an error, and the database open
+operation will be aborted. A database without a valid manifest cannot operate.
+
+#### Handling SST Corruption in Cache Files
+If SlateDB detects corruption in a cache file, it should invalidate the cache
+file and remove the corrupt file. SlateDB will not invoke `DBOptions.on_corruption`
+as there is no action for the operator to take.
+
+#### Missing Files
+When SlateDB detects a missing file in the object store that should exist, it treats this as data corruption 
+and triggers the `DBOptions.on_corruption` callback . SlateDB then takes appropriate action based on the 
+specific operation being performed when the corruption was detected. See above subsections for how each 
+operation handles corruption.
+
+#### Object Store Checksum Failures
+It is possible the underlying object store implements checksum verification when retrieving objects from the store.
+If the object store detects a checksum mismatch the `DBOptions.on_corruption` is invoked and appropriate action
+specific operation at hand is taken. See above subsections for how each operation handles corruption.
+
+When retrieving objects from the store, the underlying object store may perform checksum verification to ensure 
+data integrity. If a checksum mismatch is detected during retrieval, SlateDB invokes the `DBOptions.on_corruption`
+callback. The system then handles the corruption according to operation-specific procedures defined for each 
+type of operation. See above subsections for how each operation handles corruption. 
+
+### Golang Details
+I propose the creation of a public package called `corruption` which creates a namespace for interacting
+with database corruption.
+```go
+package corruption
+
+type Kind int
+
+// String returns the kind as a string
+func (k Kind) String() string { /*...*/ }
+
+const (
+  // A list of all the possible kinds of corruption
+  KindWAL Type = iota
+  KindSST
```

Comment:
I think `KindCompacted` is more appropriate here since WALs and compacted files are both SSTs.